### ISSUE TITLE
CATCHALL parser error when inside a pattern matching lambda

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#1383]
(This is the last issue when using --exact-split with the suite of success-full  tests)

``` bash
$ cd test/succeed/
$ agda --exact-split PatternMatchingLambda.agda
...
```

Then add `{-# CATCHALL #-}` in line 47

``` bash
$ agda --exact-split PatternMatchingLambda.agda
...
Parse error {-#<ERROR> CATCH_ALL #-} ...
```

Original issue reported on code.google.com by `andres.s...@gmail.com` on 4 Dec 2014 at 9:01

**Comment by asr** [comment]
> Here's my current attempt: jespercockx@8789c14

@jespercockx the above commit doesn't exits.

**Comment by jespercockx** [comment]
Oops, apparently I deleted that branch from my github accidentally. Here's a version of the patch that's up-to-date with the current master: https://github.com/jespercockx/agda/commit/e48a45ca97b83e8337e12b59c12bcf90f16f590a

**Comment by andreasabel** [comment]
I commented on your patch that you might have to inline `LamCatchall` in `Parser.y`.

**Comment by jespercockx** [comment]
Ah, inlining the catchall pragma into NonAbsurdLamClause and AbsurdLamClause seems to have done the trick, see https://github.com/jespercockx/agda/commit/a3a4d27dc68d04dbfb6276e5139d51002ebb7613

**Comment by andreasabel** [comment]
Good.  Teaching Programming Language Technology and Compiler Construction did give me the practical knowledge to work around LALR's non-compositionality... :)

**Comment by asr** [comment]
Testing #1729...

**Comment by asr** [comment]
Fixed by #1729.
